python/dgl/core.py

```diff
@@ -291,6 +291,17 @@ def message_passing(g, mfunc, rfunc, afunc):
             msgdata = invoke_edge_udf(g, ALL, g.canonical_etypes[0], mfunc, orig_eid=orig_eid)
         # reduce phase
         if is_builtin(rfunc):
+            # Converting msgdata from tuple of tensors to dict of tensors for heterographs
+            mdict = dict()
+            if g._graph.number_of_etypes() > 1:
+                msgdata_dict = {}
+                key = list(msgdata.keys())[0]
```

Comment:
what does msgdata contain?

Reply:
The output of message phase.

Reply:
This also means the `message_passing` function should return a value of type `Dict[str, List[Tensor]]` accordingly.